baremetal_network_provisioning/ml2/hp_network_provisioning_driver.py

```diff
@@ -119,41 +119,17 @@ def bind_port_to_segment(self, port_dict):
         external SDN controller for provisioning VLAN for the switch port where
         bare metal is connected.
         """
-        switchport = port_dict['port']['switchports']
-        switch_mac_id = switchport[0]['switch_id']
-        segmentation_id = port_dict['port']['segmentation_id']
-        bind_dict = {'neutron_port_id': port_dict['port']['id'],
-                     'access_type': hp_const.ACCESS,
-                     'segmentation_id': segmentation_id,
-                     'bind_requested': True,
-                     }
-        if switch_mac_id:
-            put_url = self._frame_port_url(switch_mac_id)
-        try:
-            port_pay_load = self._get_port_pay_load(port_dict)
-            LOG.debug("port_pay_load %(port_pay_load)s ",
-                      {'port_pay_load': port_pay_load})
-            resp = self._do_request('PUT', put_url, port_pay_load)
-            resp.raise_for_status()
-            if resp.status_code == 204:
-                LOG.debug("PUT request for physicalInterfaces is succeeded")
-                db.update_hp_ironic_swport_map_with_seg_id(self.context,
-                                                           bind_dict)
-                return hp_const.BIND_SUCCESS
-            else:
-                return hp_const.BIND_FAILURE
-        except requests.exceptions.Timeout as e:
-            LOG.error("Timed out in SDN controller : %s", e)
-            raise hp_exec.HPNetProvisioningDriverError(msg="Timed Out"
-                                                           "with SDN"
-                                                           "controller: %s"
-                                                           % e)
-        except requests.exceptions.SSLError as e:
-            LOG.error("SSLError to SDN controller : %s", e)
-            raise hp_exec.SslCertificateValidationError(msg=e)
-        except Exception as e:
-            LOG.error("ConnectionFailed to SDN controller : %s", e)
-            raise hp_exec.ConnectionFailed(msg=e)
+        LOG.debug("bind_port_to_segment with port dict %(port_dict)s",
+                  {'port_dict': port_dict})
```

Comment:
Koteswar,
front end mechanism driver provided port_dict is not aware of back-end mechanism driver needs.
For SDN controller REST input json is different from the front end provides and hence we are again reconstructing the port_dict